### PR TITLE
Psy drain, another way of getting psych points

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

All xenos gain the psy drain ability, wich can be executed on dead marine bodies. Every dead marine psy drained will give 60 pysh points, the equivalent of 2 minutes of full generator mining.

A psy drained marine is NOT perma dead, it's not even damaging him.

Silo price ranked up to 1200 from 900 to take into account this new way of getting psych points

EDIT: psy drain will also give 1/8 of a larva, on top of the psych points

## Why It's Good For The Game

Give xenos a reward to kill marines, and make them deserve their psych points. This is currently not replacing the generators psych generation, but i'll maybe nerf the generators later so psych points are aquired more actively

Also gives larva point now, because i saw some hive prefering having no silos rather than having to defend them. I think it's a neat choice, and this ability give xenos more possible strategies to experiment

## Changelog
:cl:
add: Psy drain, an ability that all xenos have that allow them to get psych points and 1/8th of a larva from dead marines. A psy drained marine is NOT perma dead, this is not headbite
balance: Silos get more expensive, to account for psy drain
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 1]
```diff
@@ -1068,6 +1067,68 @@
 	GLOB.round_statistics.xeno_rally_hive++ //statistics
 	SSblackbox.record_feedback("tally", "round_statistics", 1, "xeno_rally_hive")
 
+//*********
+// Soul mining
+//*********
+/datum/action/xeno_action/activable/soul_mining
+	name = "Soul mining"
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1074]:
```suggestion
	name = "Soul Mine"
```
Present tense over progressive tense.

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 2]
```diff
@@ -1068,6 +1067,68 @@
...
+		if(!silent)
+			to_chat(X, "<span class='warning'>This creature is struggling too much for us to aim precisely.</span>")
+		return FALSE
+	if(victim.soul_mined)
+		if(!silent)
+			to_chat(X, "<span class='warning'>This creature has already been mined for psych points.</span>")
+		return FALSE
+	if(issynth(victim)) //checks if target is a synth
+		if(!silent)
+			to_chat(X, "<span class='warning'>This creature has no soul</span>")
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1109]:
```suggestion
			to_chat(X, "<span class='warning'>This creature has no soul.</span>")
```

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 3]
```diff
@@ -1068,6 +1067,68 @@
...
+		X.visible_message("<span class='xenowarning'>\The [X] retracts its inner jaw.</span>", \
+		"<span class='danger'>We retract our inner jaw.</span>", null, 20)
+		return FALSE
+	succeed_activate() //dew it
+
+/datum/action/xeno_action/activable/soul_mining/use_ability(mob/M)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/carbon/victim = M
+
+	X.visible_message("<span class='xenodanger'>\The [X] feeds the hive with \the [victim]'s life force</span>", \
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1125]:
```suggestion
	X.visible_message("<span class='xenodanger'>\The [X] feeds the hive with \the [victim]'s life force!</span>", \
```

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 4]
```diff
@@ -1068,6 +1067,68 @@
...
+	"<span class='xenodanger'>We suddenly feel \the [victim]'s life force affluing into us</span>")
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1126]:
```suggestion
	"<span class='xenodanger'>We suddenly feel \the [victim]'s life force affluing into us!</span>")
```
What is "affluing"? Is that even a word I can find on Google?

Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:1126]: fuck, that might be a french word only. Let me check

Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:1126]: Yeah, that's a french word i poorly translated, thanks for spotting that

File: code/modules/mob/living/carbon/carbon_defines.dm [hunk 5]
```diff
@@ -45,3 +45,6 @@
 
 	/// % Chance of exploding on death, incremented by total damage taken if not initially zero.
 	var/gib_chance = 0
+
+	///TRUE if a xenos already stole life force from the mob
+	var/soul_mined = FALSE
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/carbon_defines.dm:50]: Use status effects

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request closed without merging**